### PR TITLE
Angular knobs with template

### PR DESCRIPTION
**Pull request by alterx**
Issue: #2756, #2752

## What I did
Fixed the knobs addon so it works with the new template API.

Added a fix to `@storybook/angular` core to address a bug in which a knob with `0` as a value (number) wouldn't render.

## How to test
```
yarn bootstrap --reset --core
cd examples/angular-cli
yarn storybook
open http://localhost:9008
```

### CODE REVIEW

File: addons/knobs/src/angular/helpers.js [hunk 1]
```diff
@@ -1,22 +1,19 @@
 /* eslint no-underscore-dangle: 0 */
 // eslint-disable-next-line import/no-extraneous-dependencies
 import { Component, SimpleChange, ChangeDetectorRef } from '@angular/core';
-import { getParameters, getAnnotations, getPropMetadata } from './utils';
+import { getParameters, getAnnotations } from './utils';
```
Comment (igor-dv) [inline addons/knobs/src/angular/helpers.js]: May you now delete the `getPropMetadata ` from utils ?

File: examples/angular-cli/src/stories/addon-knobs.stories.ts [hunk 2]
```diff
@@ -20,11 +20,23 @@ storiesOf('Addon Knobs', module)
...
     const phoneNumber = text('phoneNumber', '555-55-55');
 
     return {
-      component: SimpleKnobsComponent,
+      moduleMetadata: {
+        entryComponents: [SimpleKnobsComponent],
+        declarations: [SimpleKnobsComponent],
+      },
+      template: `
+        <h1> This is a template </h1>
```
Comment (igor-dv) [inline examples/angular-cli/src/stories/addon-knobs.stories.ts]: You need now update the snapshot 😅

**Review by igor-dv — APPROVED** [review]

**Review by ndelangen — APPROVED** [review]

File: examples/angular-cli/src/stories/addon-knobs.stories.ts (hunk 2 above)
Comment (alterx) [reply examples/angular-cli/src/stories/addon-knobs.stories.ts]: I tried but got this error :( 
```
yarn run v1.3.2
$ node ./scripts/test.js
? Select which tests to run Update all snapshots (update)
● Validation Error:

  Module <rootDir>/node_modules/react-native/jest/assetFileTransformer.js in the transform option was not found.

  Configuration Documentation:
  https://facebook.github.io/jest/docs/configuration.html
```

Comment (igor-dv) [reply examples/angular-cli/src/stories/addon-knobs.stories.ts]: try `node node_modules\jest\bin\jest.js angulatshots -u`

Comment (alterx) [reply examples/angular-cli/src/stories/addon-knobs.stories.ts]: `node node_modules/jest/bin/jest.js examples/angular-cli/angularshots.test.js -u` did the trick :)

Comment (igor-dv) [reply examples/angular-cli/src/stories/addon-knobs.stories.ts]: image

File: addons/knobs/src/angular/helpers.js (hunk 1 above)
Comment (alterx) [reply addons/knobs/src/angular/helpers.js]: Done

**Pull request merged**